rslaser/rspulse/pulse.py

```diff
@@ -44,59 +44,40 @@
 )
 
 
-def check_fields(input_obj, req_fields, exception, class_name):
-    for p in input_obj:
-        if p not in req_fields:
-            raise exception(f'invalid inputs: {p} is not a parameter to {class_name}')
-
-
-def _validate_params(input_params):
-    check_fields(input_params, _LASER_PULSE_DEFAULTS.keys(), InvalidLaserPulseInputError, 'LaserPulse')
-    check_fields(input_params.slice_params, _LASER_PULSE_SLICE_DEFAULTS.keys(), InvalidLaserPulseInputError, 'LaserPulseSlice')
+def check_fields(input_params, obj):
+    for p in input_params:
+        if p not in obj._DEFAULTS:
+            raise obj._INPUT_ERROR(f'invalid inputs: {p} is not a parameter to {obj.__class__}')
 
 
 def validate_type(input, _type, _class, param, exception):
     if type(input) != _type:
         raise exception(f'invalid input type: {_class} takes {param} as type:{_type} for input.')
 
 
-class LaserPulseBase:
+class LaserBase:
     '''
-        Base class for LaserPulse and LaserPulseSlice
+        Base class for LaserPulse LaserPulseSlice
         Used for input validation
     '''
-    __LASER_PULSE_DEFAULTS = _LASER_PULSE_DEFAULTS
-    __LASER_PULSE_SLICE_DEFAULTS = _LASER_PULSE_SLICE_DEFAULTS
-
-    def __init__(self, params):
-        params = self.__get_params(params)
-        return params
-
-    def __get_params(self, params):
+    def _get_params(self, params):
         if params == None:
-            return self.__LASER_PULSE_DEFAULTS.copy()
-        validate_type(params, PKDict, LaserPulseSlice, 'params', InvalidLaserPulseInputError)
-        for k in self.__LASER_PULSE_DEFAULTS:
+            return self._DEFAULTS.copy()
+        validate_type(params, PKDict, self.__class__, 'params', self._INPUT_ERROR)
```

Comment:
I think validate_type can be moved into the super class. Then you don't have to pass self.__class__ and self._INPUT_ERROR in you will be able to just get them off of self.